File: pycode/my_open_meteo_functions.py
import requests

def get_weather(lat, lng):
    url = "https://api.open-meteo.com/v1/forecast?latitude=" + str(lat) + "&longitude=" + str(lng) + "&hourly=temperature_2m,precipitation_probability,rain,weather_code"
    weather_today = requests.get(url).json()

    return(weather_today)

def code2WeatherJP(code:int) -> str:
    if   code == 0:
        return "快晴"
    elif code == 1:
        return "晴れ"
    elif code == 2:
        return "時々曇り"
    elif code == 3:
        return "曇り"
    elif code == 45:
        return "霧"
    elif code == 48:
        return "霧氷"
    elif code == 51:
        return "薄い霧雨"
    elif code == 53:
        return "霧雨"
    elif code == 55:
        return "濃い霧雨"
    elif code == 56:
        return "氷点下の薄い霧雨"
    elif code == 57:
        return "氷点下の濃い霧雨"
    elif code == 61:
        return "弱い雨"
    elif code == 63:
        return "雨"
    elif code == 65:
        return "激しい雨"
    elif code == 66:
        return "氷点下の弱い雨"
    elif code == 67:
        return "氷点下の激しい雨"
    elif code == 71:
        return "弱い雪"
    elif code == 73:
        return "雪"
    elif code == 75:
        return "激しい雪"
    elif code == 77:
        return "霧雪"
    elif code == 80:
        return "弱いにわか雨"
    elif code == 81:
        return "にわか雨"
    elif code == 82:
        return "激しいにわか雨"
    elif code == 85:
        return "弱いにわか雪"
    elif code == 86:
        return "激しいにわか雪"
    elif code == 95:
        return "雷雨"
    elif code == 96:
        return "雹を伴う雷雨"
    elif code == 99:
        return "雹を伴う激しい雷雨"
    else:
        return "未定義"

def code2Weather(code:int) -> str:
    if   code == 0:
        return "clear"
    elif code == 1:
        return "sunny"
    elif code == 2:
        return "partly cloudy"
    elif code == 3:
        return "cloudy"
    elif code == 45:
        return "fog"
    elif code == 48:
        return "freezing fog"
    elif code == 51:
        return "light drizzle"
    elif code == 53:
        return "drizzle"
    elif code == 55:
        return "dense drizzle"
    elif code == 56:
        return "light freezing drizzle"
    elif code == 57:
        return "dense freezing drizzle"
    elif code == 61:
        return "slight rain"
    elif code == 63:
        return "rain"
    elif code == 65:
        return "heavy rain"
    elif code == 66:
        return "light freezing rain"
    elif code == 67:
        return "heavy freezing rain"
    elif code == 71:
        return "slight snow falls"
    elif code == 73:
        return "snow falls"
    elif code == 75:
        return "heavy snow falls"
    elif code == 77:
        return "snow grains"
    elif code == 80:
        return "slight rain showers"
    elif code == 81:
        return "rain showers"
    elif code == 82:
        return "heavy rain showers"
    elif code == 85:
        return "slight snow showers"
    elif code == 86:
        return "heavy snow showers"
    elif code == 95:
        return "thunderstorm"
    elif code == 96:
        return "thunderstorm with slight hail"
    elif code == 99:
        return "thunderstorm with heavy hail"
    else:
        return "undefined"

def main():
    lat = 35.6895  # Example latitude for Tokyo
    lng = 139.6917  # Example longitude for Tokyo
    weather_data = get_weather(lat, lng)
    print(weather_data)

if __name__ == '__main__':
    main()

File: pycode/my_open_meteo_functions.rb
require 'net/http'
require 'json'

def get_weather(lat, lng)
  url = URI("https://api.open-meteo.com/v1/forecast?latitude=#{lat}&longitude=#{lng}&hourly=temperature_2m,precipitation_probability,rain,weather_code")
  response = Net::HTTP.get(url)
  weather_today = JSON.parse(response)
  weather_today
end

def code2WeatherJP(code)
  case code
  when 0 then "快晴"
  when 1 then "晴れ"
  when 2 then "時々曇り"
  when 3 then "曇り"
  when 45 then "霧"
  when 48 then "霧氷"
  when 51 then "薄い霧雨"
  when 53 then "霧雨"
  when 55 then "濃い霧雨"
  when 56 then "氷点下の薄い霧雨"
  when 57 then "氷点下の濃い霧雨"
  when 61 then "弱い雨"
  when 63 then "雨"
  when 65 then "激しい雨"
  when 66 then "氷点下の弱い雨"
  when 67 then "氷点下の激しい雨"
  when 71 then "弱い雪"
  when 73 then "雪"
  when 75 then "激しい雪"
  when 77 then "霧雪"
  when 80 then "弱いにわか雨"
  when 81 then "にわか雨"
  when 82 then "激しいにわか雨"
  when 85 then "弱いにわか雪"
  when 86 then "激しいにわか雪"
  when 95 then "雷雨"
  when 96 then "雹を伴う雷雨"
  when 99 then "雹を伴う激しい雷雨"
  else "未定義"
  end
end

def code2Weather(code)
  case code
  when 0 then "clear"
  when 1 then "sunny"
  when 2 then "partly cloudy"
  when 3 then "cloudy"
  when 45 then "fog"
  when 48 then "freezing fog"
  when 51 then "light drizzle"
  when 53 then "drizzle"
  when 55 then "dense drizzle"
  when 56 then "light freezing drizzle"
  when 57 then "dense freezing drizzle"
  when 61 then "slight rain"
  when 63 then "rain"
  when 65 then "heavy rain"
  when 66 then "light freezing rain"
  when 67 then "heavy freezing rain"
  when 71 then "slight snow falls"
  when 73 then "snow falls"
  when 75 then "heavy snow falls"
  when 77 then "snow grains"
  when 80 then "slight rain showers"
  when 81 then "rain showers"
  when 82 then "heavy rain showers"
  when 85 then "slight snow showers"
  when 86 then "heavy snow showers"
  when 95 then "thunderstorm"
  when 96 then "thunderstorm with slight hail"
  when 99 then "thunderstorm with heavy hail"
  else "undefined"
  end
end

def main
  lat = 35.6895  # Example latitude for Tokyo
  lng = 139.6917 # Example longitude for Tokyo
  weather_data = get_weather(lat, lng)
  puts weather_data
end

if __FILE__ == $0
  main
end